### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by JavierGSorli**
Añado la tarea de sparql y la tarea 1 de buscar un csv. Ya que borre el pull request al hacerselo a un compañero para el hands on y creo que subí archivos que no eran.

### CODE REVIEW

File: Assignment4/JavierGSorli-22C018/task06.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+"""**TASK 6.4: Add to the individual JaneSmith the email address, fullName, given and family names**"""
+
+# TO DO
+# Visualize the results
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+g.add((ns.JaneSmith, vcard.hasEmail, Literal("jane@gmail.com")))
+g.add((ns.JaneSmith, vcard.fn, Literal("Jane Smith")))
+g.add((ns.JaneSmith, vcard.hasGivenName, Literal("Jane")))
+g.add((ns.JaneSmith, vcard.hasFamilyName, Literal("Smith")))
```
Comment (dgarijo) [inline Assignment4/JavierGSorli-22C018/task06.py:65]: vcard does not have hasGivenName, hasEmail, hasFamilyName. Either use the ones defined in the vocabulary, or create your own

File: Assignment4/JavierGSorli-22C018/task07.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+ns=Namespace("http://somewhere#")
+
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE{
+    ?Subject rdfs:subClassOf ns:LivingThing.
```
Comment (dgarijo) [inline Assignment4/JavierGSorli-22C018/task07.py:32]: you are not listing all subclasses recursively

File: Assignment4/JavierGSorli-22C018/task07.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+for r in g.query(q1):
+  print(r.Subject)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE{
+    ?Subject RDF:type ns:Person.
```
Comment (dgarijo) [inline Assignment4/JavierGSorli-22C018/task07.py:47]: not listing the elements from the subclasses of person

File: Assignment4/JavierGSorli-22C018/task07.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+"""**TASK 7.3: List all individuals of "Person" or "Animal" and all their properties including their class with RDFLib and SPARQL. You do not need to list the individuals of the subclasses of person**
+
+"""
+
+q1 = prepareQuery('''
+  SELECT ?Subject ?clase ?propiedad ?valor
+  WHERE{
+    ?Subject RDF:type ?clase.
+    ?Subject ?propiedad ?valor.
+    FILTER (?clase = ns:Animal || ?clase = ns:Person)
```
Comment (dgarijo) [inline Assignment4/JavierGSorli-22C018/task07.py:65]: use UNION, not a filter which is very inefficient in this case.

File: Assignment4/JavierGSorli-22C018/task07.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+"""**TASK 7.4:  List the name of the persons who know Rocky**"""
+
+foaf=Namespace("http://xmlns.com/foaf/0.1/")
+
+q1 = prepareQuery('''
+  SELECT ?Subject
+  WHERE{
+    ?Subject foaf:knows ns:RockySmith.
+    ?Subject RDF:type ns:Person.
```
Comment (dgarijo) [inline Assignment4/JavierGSorli-22C018/task07.py:82]: you are not returning the name of the person, which is what the exercise demands

File: Assignment4/JavierGSorli-22C018/task07.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    ?Subject foaf:knows ?persona1.
+    ?Subject foaf:knows ?persona2.
+    FILTER(?persona1 != ?persona2)
+  }
+  ''',
+  initNs = {"foaf": foaf}
+)
+
+for r in g.query(q1):
+  print(r.Subject, r.persona1, r.persona2)
```
Comment (dgarijo) [inline Assignment4/JavierGSorli-22C018/task07.py:123]: Missing all exercises with rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address the requested changes

**Comment by anaigmo** [comment]
Entregado, corregido con algunos fallos, no se admiten más PR

**Pull request merged**